core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -198,12 +190,19 @@ class HadoopRDD[K, V](
     if (inputFormat.isInstanceOf[Configurable]) {
       inputFormat.asInstanceOf[Configurable].setConf(jobConf)
     }
-    val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
-    val array = new Array[Partition](inputSplits.size)
-    for (i <- 0 until inputSplits.size) {
-      array(i) = new HadoopPartition(id, i, inputSplits(i))
+    // SPARK-5068:catch the exception when the path is not exists
+    try {
+      val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
+      val array = new Array[Partition](inputSplits.size)
+      for (i <- 0 until inputSplits.size) {
+        array(i) = new HadoopPartition(id, i, inputSplits(i))
+      }
+      array
+    } catch {
+      case e: InvalidInputException => {
```

Comment:
The `HadoopRDD` is general API for reading files, I don't think it's acceptable the API just keep silence if the specified path doesn't exist, throwing an exception probably better. 

Take the JDK class `FileReader` for example:
http://docs.oracle.com/javase/7/docs/api/java/io/FileReader.html#FileReader(java.io.File)